ospd/scan.py

```diff
@@ -179,6 +179,20 @@ def set_amount_dead_hosts(self, scan_id: str, total_dead: int) -> None:
         count_dead = self.scans_table[scan_id].get('count_dead') + total_dead
         self.scans_table[scan_id]['count_dead'] = count_dead
 
+    def clean_temp_result_list(self, scan_id):
+        """ Clean the results stored in the temporary list. """
+        self.scans_table[scan_id]['temp_results'] = list()
+
+    def restore_temp_result_list(self, scan_id):
+        """ Add the results stored in the temporary list into the results
+        list again. """
+        result_aux = self.scans_table[scan_id].get('results', list())
+        result_aux.extend(self.scans_table[scan_id].get('temp_results', list()))
+
+        # Propagate results
+        self.scans_table[scan_id]['results'] = result_aux
+        self.scans_table[scan_id]['temp_results'] = list()
```

Comment:
Wouldn't it be better to call your cleanup method here too?
```suggestion
        self.scans_table[scan_id]['temp_results'] = list()
        self.clean_temp_result_list(scan_id)
```